src/main/java/org/ow2/proactive/connector/iaas/cloud/provider/azure/AzureProvider.java

```diff
@@ -87,63 +87,63 @@
  * @since 01/03/17
  */
 @Component
-public class AzureProvider implements CloudProvider {
+public abstract class AzureProvider implements CloudProvider {
 
     private static final Logger logger = Logger.getLogger(AzureProvider.class);
 
     @Getter
-    private final String type = "azure";
+    protected final String type = "azureAbstract";
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Make this final field static too. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3AS1170)